pkg/runtime/runtime.go

```diff
@@ -645,14 +645,22 @@ func (a *DaprRuntime) startHTTPServer(port, profilePort int, allowedOrigins stri
 }
 
 func (a *DaprRuntime) startGRPCInternalServer(api grpc.API, port int) error {
-	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, a.accessControlList.TrustDomain)
+	var trustDomain string
+	if a.accessControlList != nil {
+		trustDomain = a.accessControlList.TrustDomain
+	}
+	serverConf := grpc.NewServerConfig(a.runtimeConfig.ID, a.hostAddress, port, a.namespace, trustDomain)
```

Comment:
Can we extract this to share between two callers? I worry that they may drift

Reply:
I agree, also this code should go under `runtime/security`

Reply:
Will move the code later. there is some cyclic import issue to move the refactored code to runtime/security